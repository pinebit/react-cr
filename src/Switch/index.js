import React from 'react';
import PropTypes from 'prop-types';
import Wrapper from '../Wrapper';

const Switch = ({ children, index, ...wrapperProps }) => {
  const childrenCount = React.Children.count(children);

  let content = null;
  if (Array.isArray(index)) {
    content = index.filter(i => i >= 0 && i < childrenCount).map(i => children[i]);
    if (content.length === 1) {
      content = content[0];
    }
  } else {
    if (index >= childrenCount) {
      return null;
    }
    content = children[index];
  }

  return <Wrapper {...wrapperProps}>{content}</Wrapper>;
};

Switch.propTypes = {
  index: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.arrayOf(PropTypes.number)
  ]).isRequired
};

export default Switch;
