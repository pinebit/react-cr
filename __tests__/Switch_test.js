import React from 'react';
import renderer from 'react-test-renderer';
import Switch from '../src/Switch';

it('Switch renders correctly when index is valid', () => {
  const tree = renderer.create(
    <Switch index={1}>
      <h1>1</h1>
      <h1>2</h1>
      <h1>3</h1>
    </Switch>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Switch does not render when index is invalid', () => {
  const tree = renderer.create(
    <Switch index={7}>
      <h1>1</h1>
      <h1>2</h1>
      <h1>3</h1>
    </Switch>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
