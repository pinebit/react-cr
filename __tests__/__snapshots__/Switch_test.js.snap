// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch does not render when index is invalid 1`] = `null`;

exports[`Switch renders correctly when index is array 1`] = `
<div>
  <h1>
    1
  </h1>
  <h1>
    3
  </h1>
</div>
`;

exports[`Switch renders correctly when index is array with one valid index 1`] = `
<h1>
  2
</h1>
`;

exports[`Switch renders correctly when index is valid 1`] = `
<h1>
  2
</h1>
`;
